larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/ShowerTrajPointdEdx_tool.cc

```diff
@@ -158,6 +155,9 @@ namespace ShowerRecoTools{
 
     double pfpT0Time(0); // If no T0 found, assume the particle happened at trigger time (0)
     if (fT0Correct){
```

Comment:
Is this protection completely safe? It looks like a configurable member variable (does the member variable have a default initialisation to e.g. false, by the way?).

So, presumably same behaviour observed previously will be seen if tool is configured in one way, and the difference now is that it's possible to opt-out of this behaviour?

Is there a middle way where it's possible to check dynamically whether the info is available (maybe just using a try/catch block here?) and then can fail under your control, potentially more gracefully.

No need for any change if you're happy with the way that this behaves under the different configurations.

Reply:
So this variable is initialised to false by default (in [pandorashowertools.fcl](https://github.com/etyley/larpandora/blob/b3978e6cfea2152150094226070eaaf3bf0f1dc4/larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/pandorashowertools.fcl#L286)) so by default there should be no change. Indeed, the exception that is thrown currently would only be seen if this parameter is manually set to true going forward.

We could use a try catch block here but I think that throwing an exception (The title of this PR was incorrect sorry, an exception is thrown rather than a segfault) is the desired behaviour in this case as it is important for whoever is trying to run the code with T0 corrections to know that this is not going to be applied, especially so as this is an opt-in feature. 

I have also made the exception more verbose, so if anyone did stumble across the particular configurations that cause this in the future it should, hopefully, be much more clear exactly what the issue is. 

I am happy to add in a try-catch block if you still think that is best?

Reply:
Hi Ed, Good that the member variable has default initialisation. Regarding the try/catch block, I guess the question is what would you actually like to do if you catch the relevant exception? Would you be able to proceed under slightly different logic, or would you actually still want everything to bail? If there's no change in behaviour, suggest allow the exception to be caught wherever it is at the moment. If this actually bails the entire job, can ask if this is what you want (might well be the right thing to do, if the requested configuration just can't deliver). Thanks for thinking about this!

Reply:
So the exception will now only throw if the user has specifically requested to T0 correct the PFPs and this is not possible. We could just continue assuming `T0=0` (as we do in the case where `CorrectT0` is set to false) but as the user has specifically opted in to use T0s I believe we should halt the process. Especially as this is not something that we would ever expect to happen when running under normal conditions, so it should never affect production etc., but something that would only come about when running non-standard combinations of Pandora and PandoraShower which are incompatible.

Reply:
Sounds good, Ed. Thanks.